File: WarehouseWebsite.Domain/Models/Items/Item.cs
using System.Text.Json.Serialization;
using WarehouseWebsite.Domain.Models.Orders;

namespace WarehouseWebsite.Domain.Models.Items
{
    public class Item : BaseEntity
    {
        public string Name { get; set; } = null!;
        public int Quantity { get; set; }
        public string Description { get; set; } = null!;
        public decimal Price { get; set; }
        public double Weight { get; set; }
        public ItemCategory Category { get; set; }

        [JsonIgnore]
        public bool IsRemoved { get; set; }

        [JsonIgnore]
        public Guid PhotoBlobId { get; set; }

        public string? PhotoUrl { get; set; }
    }
}

File: WarehouseWebsite.Frontend/src/app/shopping/item.ts
export class Item {
  id: string;
  name: string;
  quantity: number;
  description: string;
  price: number;
  weight: number;
  category: ItemCategory;
  photoUrl: string;

  constructor(
    id: string,
    name: string,
    quantity: number,
    description: string,
    price: number,
    weight: number,
    category: ItemCategory,
    photoUrl: string) {

    this.id = id;
    this.name = name;
    this.quantity = quantity;
    this.description = description;
    this.price = price;
    this.weight = weight;
    this.category = category;
    this.photoUrl = photoUrl;
  }
}

export enum ItemCategory {
  None = "None",
  Electronics = "Electronics",
  HomeGoods = "HomeGoods",
  Clothing = "Clothing",
  BuildingMaterials = "BuildingMaterials",
  Books = "Books"
}
